### ISSUE TITLE
Can't find operator in OperatorHub after creating the catalogsource using the file based catalog

### ISSUE THREAD

**Issue by hatshepsut-working** [operator-framework/operator-registry#907]
I created a catalog image using file based catalog, applied the catalogsource, but couldn't find the operator in the OperatorHub.
However I could install the operator successfully by applying the Subscription yaml file.
I am following these docs here to build catalog image:
https://olm.operatorframework.io/docs/tasks/creating-a-catalog/#prerequisites
https://olm.operatorframework.io/docs/reference/file-based-catalogs/
The command `opm validate catalog` doesn't reporting errors for checking the catalog file.
opm version is v1.19.1, Openshift Container Platform version 4.9.5. My catalog file is below. Please help with this issue, thanks.

my-catalog-file.yaml
```

---
defaultChannel: v22.1
description: |-
  # some description here
name: my-test-operator
icon:
  base64data: 
  mediatype: 
schema: olm.package
---
image: my-bundle-image:latest
name: my-test-operator.v22.1.0
package: my-test-operator
properties:
- type: olm.gvk
  value:
    group: MyObject.ibm.com
    kind: MyObject
    version: v1
- type: olm.package
  value:
    packageName: my-test-operator
    version: 22.1.0
relatedImages:
- image: my-bundle-image:latest
  name: "my-operator-bundle"
- image: my-operator-image:latest
  name: "my-operator"
schema: olm.bundle
---
schema: olm.channel
package: my-test-operator
name: v22.1
entries:
  - name: my-test-operator.v22.1.0
---
schema: olm.bundle.object
type: "olm.bundle.object"
value: <the base64 data of csv yaml file>
```

**Comment by greglanthier** [comment]
FWIW in my early testing with file-based-catalogs I found that if the `olm.bundle.object` fields were json-ified _before_ base64 encoding them I could find my operator in OperatorHub.

For example (base64 encoding of json version of a manifest):

```yaml
    {
      "type": "olm.bundle.object",
      "value": {
        "data": "eyJhcGlWZXJzaW9uIjogInYxIiwgImRhdGEiOiB7ImNvbnRyb2xsZXJfbWFuYWdlcl9jb25maWcueWFtbCI6ICJhcGlWZXJzaW9uOiBjb250cm9sbGVyLXJ1bnRpbWUuc2lncy5rOHMuaW8vdjFhbHBoYTFcbmtpbmQ6IENvbnRyb2xsZXJNYW5hZ2VyQ29uZmlnXG5oZWFsdGg6XG4gIGhlYWx0aFByb2JlQmluZEFkZHJlc3M6IDo2Nzg5XG5tZXRyaWNzOlxuICBiaW5kQWRkcmVzczogMTI3LjAuMC4xOjgwODBcblxubGVhZGVyRWxlY3Rpb246XG4gIGxlYWRlckVsZWN0OiB0cnVlXG4gIHJlc291cmNlTmFtZTogODExYzlkYzUuZGFhcy5jb2dub3MuY2xvdWQuaWJtLmNvbVxuIn0sICJraW5kIjogIkNvbmZpZ01hcCIsICJtZXRhZGF0YSI6IHsibmFtZSI6ICJteS1vcGVyYXRvci1tYW5hZ2VyLWNvbmZpZyJ9fQ=="
      }
    }
```

instead of (base64 encoding of yaml version of a manifest):

```yaml
   {
      "type": "olm.bundle.object",
      "value": {
        "data": "YXBpVmVyc2lvbjogdjEKZGF0YToKICBjb250cm9sbGVyX21hbmFnZXJfY29uZmlnLnlhbWw6IHwKICAgIGFwaVZlcnNpb246IGNvbnRyb2xsZXItcnVudGltZS5zaWdzLms4cy5pby92MWFscGhhMQogICAga2luZDogQ29udHJvbGxlck1hbmFnZXJDb25maWcKICAgIGhlYWx0aDoKICAgICAgaGVhbHRoUHJvYmVCaW5kQWRkcmVzczogOjY3ODkKICAgIG1ldHJpY3M6CiAgICAgIGJpbmRBZGRyZXNzOiAxMjcuMC4wLjE6ODA4MAoKICAgIGxlYWRlckVsZWN0aW9uOgogICAgICBsZWFkZXJFbGVjdDogdHJ1ZQogICAgICByZXNvdXJjZU5hbWU6IDgxMWM5ZGM1LmRhYXMuY29nbm9zLmNsb3VkLmlibS5jb20Ka2luZDogQ29uZmlnTWFwCm1ldGFkYXRhOgogIG5hbWU6IG15LW9wZXJhdG9yLW1hbmFnZXItY29uZmlnCg=="
      }
    }
```

**Comment by joelanford** [comment]
This is fixed in #874.

I think we just need to cut a release.
